avocado/utils/multipath.py

```diff
@@ -0,0 +1,151 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+Module with multipath related utility functions.
+It needs root access.
+"""
+
+import time
+import logging
+from . import distro
+from . import process
+from . import service
+
+
+def form_conf_mpath_file(blacklist="", defaults_extra=""):
+    """
+    Form a multipath configuration file, and restart multipath service.
+
+    :parm blacklist: Entry in conf file to indicate blacklist section.
+    :parm defaults_extra: Extra entry in conf file in defaults section.
+    """
+    mpath_fp = open("/etc/multipath.conf", "w")
```

Comment:
Please use the `with` statement when dealing with file objects. It assures that the file is properly closed, even if an exception is raised:
```python
with open("/etc/multipath.conf", "w") as mpath_fp:
    mpath_fp.write("defaults {\n")
    ....
```

Also, the content of the file could be given to a variable and then you would call the `.write(content)` only once.